### ISSUE TITLE
How to add multiple language URLs to the XML Sitemap generated by nuxt-simple-sitemap?

### ISSUE THREAD

**Issue by msfm360** [harlan-zw/nuxt-simple-sitemap#35]
### Details

I am using nuxt-simple-sitemap for my website which has two languages, /en and /ar. However, the XML Sitemap generated by the tool only includes the default language URLs. Can someone please tell me how I can add the /en and /ar links to the XML Sitemap? 

<img width="432" alt="image" src="https://user-images.githubusercontent.com/50640440/234581560-fbf29c78-531f-4f27-9684-0e21a00d7047.png">

**Comment by msfm360** [comment]
@provok-it can you help me i saw your website have multilingual

**Comment by msfm360** [comment]
<img width="1620" alt="image" src="https://user-images.githubusercontent.com/50640440/234588158-33cf8419-8a20-4dec-89ff-e68c9503da00.png">

**Comment by provok-it** [comment]
The package is not working for me anymore since 2.4.8.

We should wait for @harlan-zw to resolve the issue first, then we'll see if you're really having a problem with it.

See #33.

**Comment by memic84** [comment]
Any idea if it's possible to add a sitemapindex, we've used the Nuxt 2 module, see their docs: https://sitemap.nuxtjs.org/usage/sitemap-index

**Comment by provok-it** [comment]
@memic84 
Read the doc and look at: https://github.com/harlan-zw/nuxt-simple-sitemap#multiple-sitemap-support

**Comment by memic84** [comment]
Yes, I read it several times and tried it, but failed to get it working. I have two dynamic sources, multiple languages and dont use the pages, and got it actually working with https://roe.dev/blog/creating-your-own-sitemap-module#a-different-approach-for-a-dynamic-sitemap

**Comment by harlan-zw** [comment]
> Any idea if it's possible to add a sitemapindex, we've used the Nuxt 2 module, see their docs: https://sitemap.nuxtjs.org/usage/sitemap-index

It should be straight forward, for example in the playground I have this config and it works without issue.

```ts
    sitemaps: {
      posts: {
        include: ['/blog/**']
      },
      pages: {
        exclude: ['/blog/**'],
      }
    }
```

If you still have issues and would like some help please create a seperate issue.

**Comment by harlan-zw** [comment]
> ### Details
> I am using nuxt-simple-sitemap for my website which has two languages, /en and /ar. However, the XML Sitemap generated by the tool only includes the default language URLs. Can someone please tell me how I can add the /en and /ar links to the XML Sitemap?
> 
> <img alt="image" width="432" src="https://user-images.githubusercontent.com/50640440/234581560-fbf29c78-531f-4f27-9684-0e21a00d7047.png">

If you're using the latest nuxt-simple-sitemap with the latest @nuxt/i18n this should _just work_ for you. You'll need to provide some reproduction details if you want some further help.

**Comment by msfm360** [comment]
> > ### Details
> > I am using nuxt-simple-sitemap for my website which has two languages, /en and /ar. However, the XML Sitemap generated by the tool only includes the default language URLs. Can someone please tell me how I can add the /en and /ar links to the XML Sitemap?
> > <img alt="image" width="432" src="https://user-images.githubusercontent.com/50640440/234581560-fbf29c78-531f-4f27-9684-0e21a00d7047.png">
> 
> If you're using the latest nuxt-simple-sitemap with the latest @nuxt/i18n this should _just work_ for you. You'll need to provide some reproduction details if you want some further help.

I'm using last version 

![image](https://user-images.githubusercontent.com/50640440/235309718-854c9d71-e7ba-409c-80ae-637765d3738a.png)

![image](https://user-images.githubusercontent.com/50640440/235309772-4bd4dd96-1277-4bc9-8f63-247bc752f9c0.png)

**Comment by harlan-zw** [comment]
Do you build your site using `nuxt build` or `nuxt generate`? Also, it looks like you're modifying the sitemap output somehow with code you haven't shared.

Ideally, you would provide a Stackblitz reproduction so I can see exactly what the setup/issue is. You can make one using the starter https://stackblitz.com/edit/nuxt-starter-dyraxc?file=package.json

**Comment by msfm360** [comment]
> Do you build your site using `nuxt build` or `nuxt generate`? Also, it looks like you're modifying the sitemap output somehow with code you haven't shared.
> 
> Ideally, you would provide a Stackblitz reproduction so I can see exactly what the setup/issue is. You can make one using the starter https://stackblitz.com/edit/nuxt-starter-dyraxc?file=package.json

thanks a lot i was using npm run build :|

now its just works fine

**Comment by lisaschumann** [comment]
I hope it is ok to post on this thread here. I am struggling to generate the dynamic routes for each page and blog post for "/en/page" and "/de/page" and would love to see how you set this up. 
I made a simple reproduction here: https://stackblitz.com/edit/nuxt-starter-uhq2jp?file=app.vue
I set the blog posts to be fetched from contentful, but here just using an array to keep it simple. Appreciate any lead you might be able to provide. I saw @provok-it also had the i18n routes working, but not sure what I am missing.

**Comment by provok-it** [comment]
@lisaschumann the Nitro prerender will generate all the internal links found from the `"/"`route.

In the sandbox env you made there is no `<NuxtLink />` tag, thus no dynamic url in your sitemap.

**Comment by lisaschumann** [comment]
@provok-it sorry I think I am not quite understanding. I moved the nav to the index page, but that's probably not what you meant. In my build I set up the nav with <NuxtLink/> to the different pages and then on the blog pages it is showing a grid of all the articles. I would like all pages as well as dynamically fetched articles to be pulled through not only as the original "/{{pagename}}" for the English version but also "/de/{{pagename}}" for the German version. I think I am missing something on the basic set-up understanding
